### DISCUSSION TITLE
Multiple account log-in

### DISCUSSION THREAD

**Discussion by Firesphere** [music-assistant/support#435] [Feature requests and ideas]
As a user, I'd like to be able to have both my own and my partner's Spotify login set up, so that we get a combined list of our Spotify playlists.

Admitted, a split option would be better, but since right now it's all flattened, I'd say "flatten it".

The playback isn't an issue, it's more about getting all the playlists etc.

**Comment by erkr** [comment]
Hi FireSphere,
If you have a Spotify premium family account, it's worth to consider one dedicated account for home assistant. 
This allows to add all playlists you want to share and prevents that Spotify complains you are using your account on two devices
Best Eric

**Reply by gtbdf1** [reply]
It would still complain, because Spotify only allows one stream per account. Which is why the ability to select which spotify account is used as source is a crucial and missing feature

**Reply by erkr** [reply]
> It would still complain, because Spotify only allows one stream per account. Which is why the ability to select which spotify account is used as source is a crucial and missing feature

AFIK MA will use the first Spotify account configured. Probably that solves you issue

**Reply by gtbdf1** [reply]
It doesn’t. Because as soon as you play Spotify in a different room, the first stream stops. Which is why selecting a source account is a crucial feature.

**Reply by erkr** [reply]
I see. I only play one stream to multiple rooms. So never encountered that limitation

**Comment by Firesphere** [comment]
Hi @erkr ,

That would be an option, but given we are on a Duo plan, since it's just the two of us, switching to a Family plan just for that feels like a bit of overkill :)

It's not a big deal, it just would be useful I think :)

**Comment by erkr** [comment]
> feels like a bit of overkill :)

Yes that is a little overkill indeed.

**Comment by marcelveldt** [comment]
The entire backend of MA is already prepared for multiple accounts. I just still need to update the HA integration config wizard.

**Reply by petwri** [reply]
Hey @marcelveldt, what is the status on the implementation?

**Reply by OzGav** [reply]
The original question was about adding multiple Spotify accounts. This is already possible.

**Reply by gtbdf1** [reply]
The original question was specifying which spotify account is used as the source, not adding multiple accounts. We still can't specify which account to use.

**Reply by distante** [reply]
> The original question was specifying which spotify account is used as the source, not adding multiple accounts. We still can't specify which account to use. 
> 

Is this now possible?

**Reply by OzGav** [reply]
No it is not

**Reply by madAndroid** [reply]
Would be amazing if that was possible, I've been using SpotCast for ages, but it keeps breaking when Spotify adjust their API

**Reply by SteveDinn** [reply]
> The entire backend of MA is already prepared for multiple accounts. I just still need to update the HA integration config wizard.

That's great to hear. I only use Subsonic, not Spotify, but I don't want music-assistant plays from the rest of my family messaing up my last.fm stats! :)  I have also been reluctant to configure my podcasts via GPodder because I don't want music-assistant plays by others to cause me to lose my place in whatever I was listening to. Multi-user environments definitely throw a wrench in things.

Home Assistant Conversation Agents now need a way to distinguish between voices of different users like Google devices can...

**Comment by erkr** [comment]
Moved to discussions, raise your votes!

**Comment by svenvdmeer** [comment]
I would like a different implementation of multiple accounts. It would be nice if I could add multiple instances of the integration that create different dashboards so me and my wife have our own Music Assistant dashboard with our own spotify account.

**Reply by TKpizza** [reply]
tis would be cool

**Reply by wjbeckett** [reply]
This would be ideal!
Main reason I haven’t told the rest of my family about MA is because of not being able to switch music accounts

**Comment by linus-ha** [comment]
I think a simple button on the bottom for switching accounts would be perfect.

We are a family of four people with completely different tastes in music and individual Spotify accounts. Would be great if all of us could use MA. I am currently running LMS, because it allows to create individual Spotify Connect devices, that we all can connect to individually from our Spotify apps.

**Reply by fhtagnn** [reply]
I'm still searching for a solution to nicely control multiple spotify accounts with different streaming endpoints to make our life easier on this end. Adding a simple button to choose the spotify account which is used to play content would make music assistant great for this purpose.

**Reply by gtbdf1** [reply]
I agree. In the meantime, I created a playlist on each kid’s account and then use music assistant to play the playlist to their respective media players. Music assistant uses the account the playlist exists in. Not ideal, but ok for now.

**Reply by fhtagnn** [reply]
Thanks for the hint. Saw that already, but for me that is not really useful. We start a lot of different audio books all the time for the kids and having individual playlists would not makt management easier ;)

**Comment by feutl** [comment]
Hi,

I am using Spotify for my kids, both have their separate accounts, but no mobile devices yet trigger or choose any music. What I do is, giving them NFC/RFID cars with a reader. So both can choose the music, audiobooks etc based on those cards.
Now, because both of them do listen to spotify at the same time but to different / same content in their rooms, I have to use different accounts.

In combination with spotify connect and the speakers I have this works, kind of with what HA already is capable of. The problem is basically choosing the speakers, if there is no mobile device running spotify, the HA media player spotify integratin does not know the speaker and therefore content cannot be played.

This seems different with how music assistant is interacting with spotify, with this integration playing music on speaker always works without problems, but only with one account.

So I would love to see music assistant having the option to choose the speaker and the spotify account the content is coming from to handle "such" scenarios.

THANKS

**Reply by OzGav** [reply]
I’m not sure of it will completely solve your issue but multiple account support is coming in the next few weeks in the next version.

**Reply by feutl** [reply]
ok, do you have any details already, why it does not solve my issue - or should I just wait until the release drops.

**Reply by OzGav** [reply]
Sorry I do not. I haven’t seen the final implementation details so can’t 100% say whether it will meet your requirements. If you could be patient that would be great.

**Reply by gtbdf1** [reply]
I still dont see the option to specify which Spotify account to play from. I was able to add all 4 spotify accounts separately, but it doesnt appear possible to tell MA to use one spotify account and play on one speaker and use a different spotify account to play on a different speaker.

**Reply by OzGav** [reply]
Playlists will play from the account they originate from

**Reply by gtbdf1** [reply]
Ok, so @feutl we should be able to create playlists in each spotify account, name them differently, and then whatever we add to the playlist will play from the correct account

**Comment by gtbdf1** [comment]
Any update on this functionality? We really need to be able to specify which Spotify account to use for which speaker.

**Reply by OzGav** [reply]
Sorry I don’t understand why it matters which Spotify account is used except for people that care about the stats that Spotify has for creating some lists but that is a separate feature request. https://github.com/orgs/music-assistant/discussions/2664

**Reply by gtbdf1** [reply]
Because Spotify only allows you to stream to one speaker at a time. So when I stream my sons sleep music to his speaker, then I stream my daughters sleep music to her speaker, it uses the same account and stops my son’s music. 

I need to be able to play from my son’s account to my son’s speaker and from my daughter’s account to my daughter’s speaker at the same time.

**Reply by fhtagnn** [reply]
Exactly, spotify wants one account per stream (at least unsing all other interfaces I know).

As I just learned music assistant seems to circumvent this somehow. Not sure if this is a TOS breaking feature, but things will still break once I use any of the official ways to control spotify or enter my car (which automatically switches a running stream on my account to it).

**Reply by gtbdf1** [reply]
Right, and a button or selector in music assistant to specify which Spotify account to use as source would fix it without breaking TOS

**Comment by TKpizza** [comment]
I can play multiple streams from one account.

**Reply by gtbdf1** [reply]
No you can’t.

**Reply by gtbdf1** [reply]
https://community.spotify.com/t5/Subscriptions/How-to-play-multiple-streams-on-different-devices/td-p/5311229

**Reply by bagobones** [reply]
It does work if all of them start IN MA (just did some testing it seems to get around some of the limits) however if you are listening on a separate Spotify device like your phone you will get kicked there so the feature request is still relevant as it would avoid conflicts outside MA.

**Comment by TKpizza** [comment]
I understand it doesn't work natively with Spotify... But if you have one
Spotify account (like I have) and multiple speakers (like I have), using
music assistant you can have different things playing on different speakers
at the SAME time. My kids are currently listening to kids lullaby from
spotify on their bedroom Google home and I am listening to music from
spotify on my living room speaker.

**Reply by gtbdf1** [reply]
Oh I see what you mean… interesting… I’ll try this tonight. Thanks

**Reply by fhtagnn** [reply]
Just tested it this morning with two devices. Not sure, if there is a limit to this, but this is clearly breaking with spotiy intended beahavior. People are asking for this as a feature on spotify forums for ages and the official answers is "one stream per account". So, maybe a TOS breaking feature?

**Reply by gtbdf1** [reply]
It didn’t work for me either. I just want a button or selector in music assistant to specify which Spotify account to use as source.

**Reply by marcelveldt** [reply]
Yeah this is a bug we need to fix some day. We however have such a big backlog that this is on the longlist.
Because if we fix this bug we also need a way to specify rules how to select an account for playback, like some sort of routing rules.
You can imagine that is not an easy thing to fix soon with our limited development team (any volunteers that want to help out here?)
Until then it is what it is, you can already stream to multiple speakers because we auto pick a free slot for streaming.

**Comment by gtbdf1** [comment]
Any chance this will be added soon? As it is, I can't play different spotify in multiple rooms at the same time, which was my main hope for MA.

**Reply by OzGav** [reply]
Sorry there is little chance for “soon” as there are a lot more higher priority items

**Reply by fhtagnn** [reply]
Too bad as this would really be a unique feature you cannot find in many other solutions.

**Reply by codemunkie15** [reply]
Surely with the increasing use of Assist, this should be higher priority? What happens if multiple people in the house want to listen to Spotify in separate rooms, at the same time?

It would be nice if you could set a default Spotify account for each user in HA, so that in the future, when voice matching is a thing, HA and MA can figure out which Spotify account to use for each person.

**Reply by marcelveldt** [reply]
> What happens if multiple people in the house want to listen to Spotify in separate rooms, at the same time?

That will just work - did you even try it ?

**Reply by Banthaf0dder** [reply]
> > What happens if multiple people in the house want to listen to Spotify in separate rooms, at the same time?
> 
> That will just work - did you even try it ?

Spotify has a very clear and explicit cap of one stream per account. If Music Assistant is able to circumvent this can you please explain how? Honestly, unless I understand how it work then I am nervous to try it and break some ToS with Spotify and get my account banned.

On top of that, I don't want to get suggestions for music based on what my kids and wife are listening to. That is another reason they have their own accounts.

**Comment by ptrnmgr** [comment]
don 't know if my request fits in this thread or i might need to raise a new one. I'm chosing this place because it might solve this issue too: a sorting option in playlists for different spotify accounts...

**Comment by bagobones** [comment]
Some routing logic would be amazing for the duo / family account houses as going to voice assistants makes this even worse.

For play lists if they are account specific it will sort of work out most of the time, unless the same playlist has been shared with another family member?

Pinning a default account to a device would help a lot as well, then I could PIN peoples room devices to their accounts and logins in HA, and the living room one could be pinned to a generic account within the family account. This would avoid most conflicts other than between my wife and I on the bedroom speaker lol.

For single song requests it would be amazing if  it did an in use check.. if the request is for a different device, used the next account not in use. This would help prevent KICKING people when just making a single song request which isn't account specific. Having an ordered list of preferred accounts for For single song requests would be needed.

**Reply by OzGav** [reply]
I think you will find some of the things above are actually not a problem when using MA. Just try it.

**Reply by bagobones** [reply]
> I think you will find some of the things above are actually not a problem when using MA. Just try it.

Interesting, if the account is in use via the official app or another device, it gets cut off as expected by MA starting a stream. However I was able to get at least 3 songs streaming WITHIN MA on 3 devices.

So if you have enough room for a FULL dedicated account for MA it is mostly good, however if you want YOUR account linked to get YOUR playlists MA will still conflict with any other Spotify device as expected.

In short some sort of streaming source / account routing  or priority logic would still be useful, or if the Spotify API supports it an IN USE on another device check.

**Reply by bagobones** [reply]
Taking this a step further if a LOCAL file source or a LOCAL source like Plex has the SAME song, it would be nice if we could say something like always prefer local copy to avoid conflicts.

**Reply by OzGav** [reply]
That is already the case https://music-assistant.io/installation/#usage-and-notes

**Reply by bagobones** [reply]
> That is already the case https://music-assistant.io/installation/#usage-and-notes

the wording on this is confusing, does the local copy need to be higher quality? What if Spotify is higher?

>"If a song is available on multiple providers (e.g. Spotify and a flac file on disk), the file/stream with the highest quality is always preferred when starting a stream. Highest quality is based on sample rate, bit depth and codec and local is preferred over cloud"

Edit: Is plex always considered local ? or is this LOCAL as in the MA native local mapping of a folder?

**Reply by bagobones** [reply]
Related to one of the issues above is it possible to BLOCK casting to devices that have a state of "External Source Active"?

**Reply by OzGav** [reply]
Maybe you could do it as part of an automation but I haven’t looked at the attributes to see what might be useful in that situation.

**Reply by bagobones** [reply]
Just going to sum this up as the request still stands:

1. MA is able to play more than one stream at a time (which is very nice)
2. MA however WILL interrupt any "other" Spotify devices using the same account such as a phone. Which is still a problem.
3. MA will also override an already playing local device which could be annoying for announcements if a user had cast from their phone to a device NOT MA and an announcement killed their stream.

**Comment by gtbdf1** [comment]
Until we can specify which Spotify account to stream from, this is a problem with MA. It’s impossible to play Spotify in two different rooms at once.

**Reply by OzGav** [reply]
I don’t know why you don’t listen to the responses you get. Have you tried it?

**Reply by Banthaf0dder** [reply]
> I don’t know why you don’t listen to the responses you get. Have you tried it?

Can you explain how Music Assistant can supposedly do this? Spotify has a clear and explicit cap of one stream per account. How would I know I'm not breaking ToS with Spotify if I use Music Assistant to run more than one stream at the same time?

**Reply by Coxeroni** [reply]
I am not sure how MA does it, but as a test I could happily play three different tracks on three different devices, all with one Spotify account registered in MA.

**Reply by bagobones** [reply]
Yes it does, but if you use the normal Spotify app outside of MA then MA and all devices using that account will get interrupted and visa versa.

**Reply by Coxeroni** [reply]
Not for me, at least currently. I haven't used MA a lot, but additionally to the MA playback I can play an additional stream with the Spotify app without breaking MA playback.

**Comment by gtbdf1** [comment]
Yes, constantly. I tell MA to play Spotify on one speaker, then tell it to
play a different song on Spotify on another speaker, and it stops playing
the first song on the first speaker. This would not be an issue if we could
tell it which Spotify account to use.

**Reply by OzGav** [reply]
Then you have something wrong with your setup

**Comment by Cezero** [comment]
Adding my vote for this feature. I have Spotify family and the ability to stream from different accounts is crucial for my use case as I am often streaming in my car or on my phone which interrupts anyone streaming in via MA. I have 6 Spotify accounts, each user should be able to stream from their own account in MA. Thank you.

**Reply by eebiec** [reply]
seriously - this is so important. i dont know why the MA folks keep telling us this is unnecessary

**Reply by marcelveldt** [reply]
Not sure what you are referring to but this feature exists today

**Comment by Cezero** [comment]
From the maintainer responses on this thread it is clear to me that there is a bit of a misunderstanding regarding what is being requested vs what is currently available.

To summarize, what people are asking for is the ability to register multiple Spotify accounts with Music Assistant, and then have the ability to determine which of those accounts is used to play music for any given stream. That would allow for the scenario where my wife wants to listen to Spotify music on her account from Music Assistant, in the living room, while I am off running an errand and listening to Spotify in my car on my account. Then when I get home, if I want to have Music Assistant stream from my account into the office, that would clearly interrupt the "stream to my car" as it's the same account, but would *not* interrupt my wife's streaming via her account in the living room.

In my case, I have a family subscription to Spotify with 5 accounts (myself, my wife and our three daughters). My daughters often stream Spotify to speakers in their bedrooms while I am listening to my music in the office or the workshop. Each of these streams should be capable of being associated with the appropriate, different, Spotify account. I don't want my daughter's poor music tastes influencing my Spotify recommendations, nor do I want to interrupt their streams if I start playing Spotify outside of Music Assistant on my phone or another device.

**Reply by bagobones** [reply]
Spotify family here as well same issue. I work around it by having a generic account use one of the family slots but then we don't get any custom playlists per user on the speaker devices .

Having some routing for multiple accounts would be really nice where you could make some bedrooms / devices defaulted to a specific account etc or have a prompt to select an account on the fly.

**Reply by PatrickGlatz** [reply]
Hello DEV Team,

also our family is using a spotify family account and each one has his own account, which are separated perfectly in MusicAssistant. But in HomeAssistant using the default media-control card or the custom:mediocre-media-control-card, it is not possible to specify which account should be used for e.g. searching, displaying of playlists and for the play count number assignment (we all wait for our spotify wrapped at the end of the year, don't we?). So account separation also for the HomeAssistant interaction (like with the media-control card) would be really needed.

Could you please give a short update?

**Comment by Tapperst** [comment]
To switch between Accounts would be nice! 
Is it possible to have more music assistant instances in home assistant?
Each instance has it own spotify account. Then you have 5 times the same music assistant media player for each speaker but with different accounts. Is this possible?

**Reply by OzGav** [reply]
Not as an addon but you could run mutliple docker containers

**Reply by Tapperst** [reply]
You mean multiple home assistant instances in docker containers or this addon? How does it work?

**Reply by OzGav** [reply]
No you don’t use the add on method you setup up multiple standalone MA instances in their own docker containers. I don’t think you would add them all to HA but you could I guess. People would do this to separate out libraries for different people.

**Reply by felixschndr** [reply]
I thought about this workaround as well. Not pretty since everything else like the player providers has to be duplicated but 
> mutliple docker containers

seems to be only way at the moment

**Comment by gtbdf1** [comment]
I don’t understand why the devs can set it up so allow multiple Spotify
accounts so we can select different accounts for different speakers. What
is the limitation preventing this?

**Reply by eebiec** [reply]
it feels like the limitation is that they don't think it's important / don't care / don't want to. :( from their replies.

**Comment by processor286** [comment]
I had thought I'd found a way to get our Spotify Family to work with Home Pods via Music Assistant but we have the same problem.

A solution would seem to me to either

1) have multiple Spotify providers available in Music Assistant, and a way to chose which provider is utilised for each stream, so four people can use their own account to stream to a Home Pod.  

or 

2) Allow multiple installations of Music Assistant into HA, and allow HA to choose which instance of MA to be used based on some criteria. 

or 

3) ??? some other way I've not thought of

Hopefully I've described the requirement sufficiently to foster a productive outcome

**Reply by processor286** [reply]
Wait hang on, I can run multiple MA containers? I don't know how I'd do that on Home Assistant Green, but I'll think on it. I might also be able to deploy some MA containers elsewhere, presuming they don't have to be on the same host. 

Is this is what is meant by the discussion having been 'implemented' ?

**Reply by OzGav** [reply]
You can already do your option 1 by adding all of the Spotify accounts to MA and then restrict users to a certain one https://www.music-assistant.io/settings/user-management/

**Reply by processor286** [reply]
Oh right wow, sorry for adding noise. Thanks

**Reply by felixschndr** [reply]
> You can already do your option 1 by adding all of the Spotify accounts to MA and then restrict users to a certain one https://www.music-assistant.io/settings/user-management/

This sounds good and would help a lot.
I am wondering whether it is possible to link accounts to voice pipelines. With this two people could use different wake words which would trigger different pipelines. This would then choose the Spotify account to play from.

**Reply by OzGav** [reply]
No as the HA actions can’t specify providers currently

**Reply by felixschndr** [reply]
Are there any plans to change that? @OzGav

**Reply by OzGav** [reply]
Not at the moment. It could be fairly complex. I suggest you open a feature request and we can see how many votes it gets.

**Reply by felixschndr** [reply]
will do

**Reply by felixschndr** [reply]
Done in https://github.com/orgs/home-assistant/discussions/3337

**Reply by bagobones** [reply]
I would less prefer to do it by wake word and more prefer to do it by device. IE each person's bedroom device uses their spotify provider by default and the common area ones use a generic account or something.

Ether way however it would require there to be an api or service call to make it specific

**Comment by gtbdf1** [comment]
It’s not, really. The only real way is to run multiple instances of MA,
which isn’t exactly preferred.

**Reply by OzGav** [reply]
What's "not really"?

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
